code/game/objects/structures/guardrail.dm

```diff
@@ -0,0 +1,41 @@
+/obj/structure/guardrail
+	name = "guard rails"
+	icon = 'icons/obj/doors/railing.dmi'
+	icon_state = "railing1"
+	flags_atom = ON_BORDER
+	opacity = FALSE
+	density = TRUE
+	throwpass = FALSE
+	layer = WINDOW_LAYER
+	anchored = TRUE
+	coverage = 25
+	resistance_flags = UNACIDABLE
+	max_integrity = 150
+	soft_armor = list("melee" = 90, "bullet" = 30, "laser" = 30, "energy" = 30, "bomb" = 15, "bio" = 100, "rad" = 100, "fire" = 100, "acid" = 100)
+	var/closed = TRUE
+	var/is_wired = TRUE
+
+/obj/structure/guardrail/CheckExit(atom/movable/O, turf/target)
+	if(closed)
+		return TRUE
```

Comment:
is this copypaste
it feels like copypaste

Reply:
Sure is!

Reply:
It's ugly, and copypaste, but it works.